mat64/dense.go

```diff
@@ -512,7 +512,13 @@ func (m *Dense) UnmarshalBinary(data []byte) error {
 	m.mat.Stride = int(cols)
 	m.capRows = int(rows)
 	m.capCols = int(cols)
-	m.mat.Data = use(m.mat.Data, m.mat.Rows*m.mat.Cols)
+
```

Comment:
why would you artificially limit the number of elements an array/slice/vector could hold?
if you bought a high-end machine with loads of memory, it's to be used :)

unless you disagree, I'll keep the check on the number of elements in the slice.